test/prefer-t-regex.js

```diff
@@ -61,6 +63,42 @@ ruleTester.run('prefer-t-regex', rule, {
 			code: header + 'const reg = /\\d+/;\ntest(t => t.true(reg.test(foo.bar())));',
 			output: header + 'const reg = /\\d+/;\ntest(t => t.regex(foo.bar(), reg));',
 			errors: errors('regex')
+		},
+		// The same as the above tests but with `RegExp()` object instead of a regex literal
+		{
+			code: header + 'test(t => t.true(new RegExp("\\d+").test("foo")));',
+			output: header + 'test(t => t.regex("foo", new RegExp("\\d+")));',
+			errors: errors('regex')
+		},
+		{
+			code: header + 'test(t => t.false(foo.search(new RegExp("\\d+"))));',
+			output: header + 'test(t => t.notRegex(foo, new RegExp("\\d+")));',
+			errors: errors('notRegex')
+		},
+		{
+			code: header + 'const regexp = RegExp("\\d+");\ntest(t => t.true(foo.search(regexp)));',
+			output: header + 'const regexp = RegExp("\\d+");\ntest(t => t.regex(foo, regexp));',
+			errors: errors('regex')
+		},
+		{
+			code: header + 'test(t => t.truthy(foo.match(new RegExp("\\d+"))));',
+			output: header + 'test(t => t.regex(foo, new RegExp("\\d+")));',
+			errors: errors('regex')
+		},
+		{
+			code: header + 'test(t => t.false(RegExp("\\d+").test("foo")));',
+			output: header + 'test(t => t.notRegex("foo", RegExp("\\d+")));',
+			errors: errors('notRegex')
+		},
+		{
+			code: header + 'test(t => t.true(new RegExp("\\d+").test(foo())));',
+			output: header + 'test(t => t.regex(foo(), new RegExp("\\d+")));',
+			errors: errors('regex')
+		},
+		{
+			code: header + 'const reg = RegExp("\\d+");\ntest(t => t.true(reg.test(foo.bar())));',
+			output: header + 'const reg = RegExp("\\d+");\ntest(t => t.regex(foo.bar(), reg));',
```

Comment:
`RegExp` can also accept a regex literal. That needs to be tested as well: `RegExp(/foo/)`.